redis cache missing args_mapping causes panic

**Problem**

Missing `args_mapping` from redis processor causes panic during runtime.

**Example**

```yaml
input:
   stdin: {}

pipeline:
  processors:
    - redis:
         url: redis://localhost:6379
         command: foo

output:
   stdout: {}
```

**Result**

```shell
❯ ./target/bin/benthos -c config.yaml
INFO Running main config from specified file       @service=benthos path=config.yaml
INFO Launching a benthos instance, use CTRL+C to close  @service=benthos
INFO Listening for HTTP requests at: http://0.0.0.0:4195  @service=benthos
{"foo":"foobar"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a556e7]

goroutine 40 [running]:
github.com/benthosdev/benthos/v4/public/service.MessageBatch.BloblangQuery({0xc001188cd0, 0x1, 0xc001586ff0?}, 0x110d4c6?, 0x11f2a10?)
	/Users/___/benthos/public/service/message.go:276 +0x27
github.com/benthosdev/benthos/v4/internal/impl/redis.(*redisProc).execRaw(0xc0015865a0, {0x454bd08, 0xc000f1e000}, 0x8?, {0xc001188cd0, 0x1, 0x1}, 0xc001188ce8)
	/Users/___/benthos/internal/impl/redis/processor.go:308 +0x78
github.com/benthosdev/benthos/v4/internal/impl/redis.(*redisProc).ProcessBatch(0xc0015865a0, {0x454bd08, 0xc000f1e000}, {0xc001188cd0?, 0x1, 0x1})
	/Users/___/benthos/internal/impl/redis/processor.go:362 +0x4d0
github.com/benthosdev/benthos/v4/public/service.(*airGapBatchProcessor).ProcessBatch(0xc000d3a1e0, {0x454bd08, 0xc000f1e000}, {0x5?, 0xc000014078?, 0x1?}, {0xc000014078, 0x1, 0x1})
	/Users/___/benthos/public/service/processor.go:109 +0xee
github.com/benthosdev/benthos/v4/internal/component/processor.(*v2BatchedToV1Processor).ProcessBatch(0xc0001a1050, {0x454bd08, 0xc000f1e000}, {0xc000014078?, 0x1, 0x1})
	/Users/___/benthos/internal/component/processor/processor_v2.go:146 +0x157
github.com/benthosdev/benthos/v4/internal/component/processor.ExecuteAll({0x454bd08, 0xc000f1e000}, {0xc000bc7140, 0x1, 0x0?}, {0xc000c84ec0?, 0x1, 0x0?})
	/Users/___/benthos/internal/component/processor/execute.go:22 +0x1b6
github.com/benthosdev/benthos/v4/internal/pipeline.(*Processor).loop(0xc000b2cf00)
	/Users/___/benthos/internal/pipeline/processor.go:69 +0x285
created by github.com/benthosdev/benthos/v4/internal/pipeline.(*Processor).Consume
	/Users/___/benthos/internal/pipeline/processor.go:139 +0x79
```

**Version**

`v4.4.1-73-gf8b1fd9d`

**Possible Fix**

Remove Default value

```diff
diff --git a/internal/impl/redis/processor.go b/internal/impl/redis/processor.go
index 4443e8f4..bc353280 100644
--- a/internal/impl/redis/processor.go
+++ b/internal/impl/redis/processor.go
@@ -36,8 +36,7 @@ performed for each message and the message contents are replaced with the result
                        Description("A [Bloblang mapping](/docs/guides/bloblang/about) which should evaluate to an array of values matching in size to the number of arguments required for the specified Redis comm>
                        Version("4.3.0").
                        Example("root = [ this.key ]").
-                       Example(`root = [ meta("kafka_key"), this.count ]`).
-                       Default(``)).
+                       Example(`root = [ meta("kafka_key"), this.count ]`)).
                Field(service.NewStringAnnotatedEnumField("operator", map[string]string{
                        "keys":   `Returns an array of strings containing all the keys that match the pattern specified by the ` + "`key` field" + `.`,
                        "scard":  `Returns the cardinality of a set, or ` + "`0`" + ` if the key does not exist.`,
```

--- a/internal/impl/redis/processor.go
+++ b/internal/impl/redis/processor.go
@@ -3,6 +3,7 @@ package redis
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"strconv"
 	"time"
@@ -35,9 +36,9 @@ performed for each message and the message contents are replaced with the result
 		Field(service.NewBloblangField("args_mapping").
 			Description("A [Bloblang mapping](/docs/guides/bloblang/about) which should evaluate to an array of values matching in size to the number of arguments required for the specified Redis command.").
 			Version("4.3.0").
+			Optional().
 			Example("root = [ this.key ]").
-			Example(`root = [ meta("kafka_key"), this.count ]`).
-			Default(``)).
+			Example(`root = [ meta("kafka_key"), this.count ]`)).
 		Field(service.NewStringAnnotatedEnumField("operator", map[string]string{
 			"keys":   `Returns an array of strings containing all the keys that match the pattern specified by the ` + "`key` field" + `.`,
 			"scard":  `Returns the cardinality of a set, or ` + "`0`" + ` if the key does not exist.`,
@@ -59,11 +60,10 @@ performed for each message and the message contents are replaced with the result
 			Description("The time to wait before consecutive retry attempts.").
 			Default("500ms").
 			Advanced()).
-		LintRule(`
-root = if this.contains("operator") && this.contains("command") {
-  [ "only one of 'operator' (old style) or 'command' (new style) fields should be specified" ]
-}
-`).
+		LintRule(`root = match {
+  this.exists("operator") && this.command != "" => [ "only one of 'operator' (old style) or 'command' (new style) fields should be specified" ]
+  this.exists("args_mapping") == this.exists("operator") => [ "either args_mapping or operator must be set" ],
+}`).
 		Example("Querying Cardinality",
 			`If given payloads containing a metadata field `+"`set_key`"+` it's possible to query and store the cardinality of the set for each message using a `+"[`branch` processor](/docs/components/processors/branch)"+` in order to augment rather than replace the message contents:`,
 			`
@@ -159,14 +159,31 @@ func newRedisProcFromConfig(conf *service.ParsedConfig, res *service.Resources) 
 	}
 
 	var argsMapping *bloblang.Executor
-	if testStr, _ := conf.FieldString("args_mapping"); testStr != "" {
+	if conf.Contains("args_mapping") {
 		if argsMapping, err = conf.FieldBloblang("args_mapping"); err != nil {
 			return nil, err
 		}
 	}
 
+	var operator redisOperator
+	if conf.Contains("operator") {
+		operatorStr, err := conf.FieldString("operator")
+		if err != nil {
+			return nil, err
+		}
+		if operator, err = getRedisOperator(operatorStr); err != nil {
+			return nil, err
+		}
+	}
+
+	if (argsMapping == nil) != (operator == nil) {
+		return nil, errors.New("either args_mapping or operator must be set")
+	}
+
 	r := &redisProc{
 		log: res.Logger(),
+
+		operator: operator,
 
 		command:     command,
 		argsMapping: argsMapping,
@@ -178,16 +195,6 @@ func newRedisProcFromConfig(conf *service.ParsedConfig, res *service.Resources) 
 
 	if conf.Contains("key") {
 		if r.key, err = conf.FieldInterpolatedString("key"); err != nil {
-			return nil, err
-		}
-	}
-
-	if conf.Contains("operator") {
-		operatorStr, err := conf.FieldString("operator")
-		if err != nil {
-			return nil, err
-		}
-		if r.operator, err = getRedisOperator(operatorStr); err != nil {
 			return nil, err
 		}
 	}

--- a/website/docs/components/processors/redis.md
+++ b/website/docs/components/processors/redis.md
@@ -330,7 +330,6 @@ A [Bloblang mapping](/docs/guides/bloblang/about) which should evaluate to an ar
 
 
 Type: `string`  
-Default: `""`  
 Requires version 4.3.0 or newer  
 
 ```yml

